okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java

```diff
@@ -76,6 +77,13 @@
       IOException ioException = new IOException("Exception in connect");
       ioException.initCause(e);
       throw ioException;
+    } catch (ClassCastException e) {
+      // On android 8.0, socket.connect throws a ClassCastException due to a bug
+      if (Build.VERSION.SDK_INT == 26) {
+        log(WARN, "Swallowed ClassCastException, see https://issuetracker.google.com/issues/63649622", e);
```

Comment:
Without seeing this exact Android code, it seems hard to review.  But I don't think this is correct based on previous versions.  It seems like the CCE is thrown while processing a root IOException.  I'm basing this on connectDetail always going into an Exception.

At minimum swallowing only seems correct if the connect is successful (although later operations will fail). 

@JakeWharton do you know anyone who could check the exact state after this exception?

Reply:
@yschimke https://android.googlesource.com/platform/libcore/+/oreo-release/luni/src/main/java/libcore/io/IoBridge.java#700 there is the Android code.

Reply:
@yschimke and for reference, there is the fixed version of the Android code, which did not make it to the initial release: https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/libcore/io/IoBridge.java

Reply:
tl;dr: rethrowing the exception as IOException seems about right.